src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -862,7 +862,9 @@ def _visit_with_horizontal(
     def _are_intervals_nonoverlapping(self, compute_blocks: List[gt_ir.ComputationBlock]):
         for i, block in enumerate(compute_blocks[1:]):
             other = compute_blocks[i]
-            if not block.interval.disjoint_from(other.interval):
+            if block.interval != other.interval and not block.interval.disjoint_from(
```

Comment:
I see the point, but we shouldn't do this as the following should still be illegal:
```python
with computation(PARALLEL):
    with interval(0, 3):
        field = 0
    with interval(0, 3):
        field = 1
```
Instead we should check if either the intervals are disjoint or the horizontal regions are disjoint.